src/nvim/ui_client.c

```diff
@@ -284,6 +285,59 @@ void ui_client_event_raw_line(GridLineEvent *g)
                (const schar_T *)grid_line_buf_char, grid_line_buf_attr);
 }
 
+/// Restarts the embedded server without killing the UI.
+void ui_client_event_restart(Array args)
+{
+  // 1. Client-side server detach.
+  ui_client_detach();
```

Comment:
What happens if multiple UIs are attached? The server should only send the event to the "current UI".